stackrox/acs-fleet-manager
# ROX-12921: initialize counter metrics with 0 value

## Description
The focus of this PR is to introduce a 0 value as a default value for "slow" counter metrics.
"Slow" here means that the counter metric is not actively updating and can take a long time before being incremented for the first time.
The reason to do that is that to correctly use the Prometheus `rate` function in the alert, a zero value should be defined. `rate` function is used to determine when the counter metric goes up.

This PR does not solve the described issue for the central timeout metric. The reason for that is that the solution within this PR works only for metrics without dynamically changing labels.  The issue for the timeout metric will be fixed on the alert side.

## Checklist (Definition of Done)


## Test manual

1. Deploy local fleet-manager
2. Open `http://localhost:8080/metrics` and see worker counter metrics and operations counter metrics:
```
# HELP acs_fleet_manager_reconciler_failure_count count of failed operations of the backgroup reconcilers
# TYPE acs_fleet_manager_reconciler_failure_count counter
acs_fleet_manager_reconciler_failure_count{worker_type="accepted_dinosaur"} 0
acs_fleet_manager_reconciler_failure_count{worker_type="central_auth_config"} 0
acs_fleet_manager_reconciler_failure_count{worker_type="deleting_dinosaur"} 0
acs_fleet_manager_reconciler_failure_count{worker_type="dinosaur_dns"} 0
acs_fleet_manager_reconciler_failure_count{worker_type="preparing_dinosaur"} 0
acs_fleet_manager_reconciler_failure_count{worker_type="provisioning_dinosaur"} 0
acs_fleet_manager_reconciler_failure_count{worker_type="ready_dinosaur"} 0
```
and
```
# HELP acs_fleet_manager_central_operations_success_count number of successful central operations
# TYPE acs_fleet_manager_central_operations_success_count counter
acs_fleet_manager_central_operations_success_count{operation="create"} 0
acs_fleet_manager_central_operations_success_count{operation="delete"} 0
# HELP acs_fleet_manager_central_operations_total_count number of total central operations
# TYPE acs_fleet_manager_central_operations_total_count counter
acs_fleet_manager_central_operations_total_count{operation="create"} 0
acs_fleet_manager_central_operations_total_count{operation="delete"} 0
```
